alacritty/src/display/mod.rs

```diff
@@ -1010,44 +1046,155 @@ impl Display {
         bar_text
     }
 
-    /// Draw current search regex.
-    fn draw_search(&mut self, config: &UiConfig, size_info: &SizeInfo, text: &str) {
-        let glyph_cache = &mut self.glyph_cache;
-        let num_cols = size_info.columns();
+    /// Format hyperlink preview to make it fit into the terminal.
+    fn format_hyperlink_preview(text: &str, max_chars: usize) -> Vec<char> {
```

Comment:
I've refactored search and hyperlink to use the same function for both of them. I do like the search code a bit more now since it's much easier to read, compared to random truncation math we had before.